numpy/lib/format.py

```diff
@@ -313,6 +313,9 @@ def descr_to_dtype(descr):
 
```

Comment:
I'd suggest replacing the line above this with:
```python
title, name = name if isinstance(name, tuple) else (None, name)
names.append(name)
formats.append(dt)
titles.append(title)
offsets.append(offset)
```
and then removing the four lines before the return in this function. You'll want to replace `fields = []` with something like
`names, formats, titles, offsets = [], [], [], []`.

Reply:
I actually tried exactly the same code. Then I found that is returns exactly same thing as `np.dtype([])`. So I came here and showed you the result of [this snippet](https://github.com/numpy/numpy/pull/15397#discussion_r370006812).

I will push the new code. You can decide which one to use

Reply:
The test you're missing is `descr_to_dtype([('', 'V8')])`, which produced the wrong result with your original code.